tests/bluetooth/tester/src/gatt.c

```diff
@@ -859,6 +861,40 @@ static void set_enc_key_size(u8_t *data, u16_t len)
 		   status);
 }
 
+static void read_local_attr_value_rp(u8_t *value, u16_t value_len)
+{
+	u8_t buf[512 + sizeof(struct gatt_read_local_attr_rp)];
+	struct gatt_read_local_attr_rp *rp = (void *) buf;
+
+	/* Maximum attribute value size is 512 bytes */
+	assert(value_len < 512);
+
+	rp->value_length = sys_cpu_to_le16(value_len);
+	memcpy(rp->value, value, value_len);
+
+	tester_send(BTP_SERVICE_ID_GATT, GATT_READ_LOCAL_ATTR_VAL,
+		    CONTROLLER_INDEX, (u8_t *) rp,
+		    sizeof(*rp) + rp->value_length);
+}
+
+static void read_local_attr_value(u8_t *data, u16_t len)
+{
+	const struct gatt_read_local_attr_cmd *cmd = (void *) data;
+	struct bt_gatt_attr *a;
+
+	for (a = server_db; a <= LAST_DB_ATTR; a = NEXT_DB_ATTR(a)) {
+		const struct gatt_value *value = a->user_data;
+
+		if (a->handle == sys_le16_to_cpu(cmd->handle)) {
```

Comment:
I would do the sys_le16_to_cpu() conversion outside of the loop so it's only done once, i.e. use a helper 'u16_t handle' variable for it.

Reply:
You have absolute right! However I changed my mind. I think that BTP event will be a more appropriate in this case. New value will be indicated, when PTS performed write operation and attribute value has been changed.

Reply:
Yep sounds much better to notify the value had changed so we don't have to guess when to send a new command to retrieve the value. Btw, aren't the writes going to the host side anyway or they are always performed locally?

Reply:
PTS initiated writes are performed on tester application's database, and were not notified to the AutoPTS. It was not necessary, because PTS checked if the write operation was performed successfully by reading the updated attribute value. However I noticed, that in some tests was added new window to confirm if the write was performed successfully. So I have to add a verification for those tests.

Reply:
Implemented event:
https://github.com/zephyrproject-rtos/zephyr/pull/4234